kayenta-core/src/main/java/com/netflix/kayenta/util/Retry.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.util;
+
+import java.util.function.Supplier;
+
+public class Retry {
```

Comment:
Please have a look at [spring-retry](https://github.com/spring-projects/spring-retry).
It might be a good solution in case there is a need for more sophisticated retry policies: like configuring which types of exceptions should be retried, adding retry metrics or providing different backoff policies.